File: Client/src/main/java/org/runite/jagex/Node.java
//package org.runite.jagex;
//
//import org.rs09.client.Linkable;
//
//class Node extends Linkable {
//
//    long nodeKey;
//    Node previousNode;
//    Node nextNode;
//
//    final void unlinkNode() {
//        if (this.previousNode != null) {
//            this.previousNode.nextNode = this.nextNode;
//            this.nextNode.previousNode = this.previousNode;
//            this.nextNode = null;
//            this.previousNode = null;
//        }
//    }
//
//}

File: Client/src/main/kotlin/org/rs09/client/Node.kt
package org.rs09.client

open class Node : Linkable() {
    @JvmField
    var nodeKey: Long = 0
    @JvmField
    var previousNode: Node? = null
    @JvmField
    var nextNode: Node? = null

    fun unlinkNode() {
        if (previousNode != null) {
            previousNode!!.nextNode = nextNode
            nextNode!!.previousNode = previousNode
            nextNode = null
            previousNode = null
        }
    }
}
